gbpservice/nfp/configurator/lib/demuxer.py

```diff
@@ -105,45 +103,56 @@ def get_service_agent_info(self, operation, service_type, request_data):
+                      const.LOADBALANCER: const.HAPROXY}
+
+        service_vendor = request_data['info']['service_vendor']
+        if str(service_vendor) == 'None':
+            service_vendor = vendor_map[service_type]
 
         for config_data in request_data['config']:
             sa_info = {}
-            if service_type in constants.supported_service_types:
-                sa_info.update({'service_type': service_type})
-                if service_type == 'firewall':
-                    method = operation + '_' + config_data['resource']
-                elif service_type == 'vpn':
-                    method = 'vpnservice_updated'
-                elif service_type == 'loadbalancer':
-                    method = operation + '_' + config_data['resource']
-                elif service_type == 'config_script':
-                    method = 'run' + '_' + service_type
-            else:
-                sa_info.update({'service_type': 'generic'})
-                if operation == 'create':
-                    method = 'configure_' + config_data['resource']
-                elif operation == 'delete':
-                    method = 'clear_' + config_data['resource']
-                elif operation == 'update':
-                    method = 'update_' + config_data['resource']
-                else:
-                    return None
-
-            sa_info.update({'method': method})
-            sa_info.update({'resource': config_data['resource']})
-
-            data = config_data['kwargs']
+
+            resource_map = {
+                'firewall': (operation + '_' + config_data['resource']),
+                'vpn': ('vpnservice_updated'),
+                'loadbalancer': (operation + '_' + config_data['resource']),
+                'nfp_service': ('run' + '_' + const.NFP_SERVICE),
+                'generic_config': {
+                           'create': ('configure_' + config_data['resource']),
+                           'update': ('update_' + config_data['resource']),
+                           'delete': ('clear_' + config_data['resource'])}}
+
+            context = request_data['info']['context']
+
+            data = config_data['resource_data']
             if not data:
                 return None
 
-            context = config_data['kwargs']['context']
-            sa_info.update({'context': context})
-            del config_data['kwargs']['context']
-            if ('generic' in sa_info['service_type'] or
-                    'config_script' in sa_info['service_type']):
-                sa_info.update({'kwargs': {'kwargs': data}})
+            resource = config_data['resource']
+            is_nfp_svc = True if resource in const.NFP_SERVICE_LIST else False
+
+            if is_generic_config:
+                method = resource_map[const.GENERIC_CONFIG][operation]
             else:
-                sa_info.update({'kwargs': data})
+                if is_nfp_svc:
+                    service_type = const.NFP_SERVICE
```

Comment:
In API, we have only 3 service types (FW/LB/VPN). However, in this code, we are filling others. Can we name this field differently, say resource_type rather than service_type?